service/worker/migration/activities.go

```diff
@@ -482,3 +540,194 @@ func (a *activities) SeedReplicationQueueWithUserDataEntries(ctx context.Context
 		activity.RecordHeartbeat(ctx, heartbeatDetails)
 	}
 }
+
+func (a *activities) createReplicationTasks(ctx context.Context, request *genearteAndVerifyReplicationTasksRequest, detail *replicationTasksHeartbeatDetails) error {
+	rateLimiter := quotas.NewRateLimiter(request.RPS, int(math.Ceil(request.RPS)))
+
+	for i := 0; i < len(request.Executions); i++ {
+		r := &detail.Results[i]
+		if r.isCompleted() {
+			continue
+		}
+
+		we := request.Executions[i]
+		tags := []tag.Tag{tag.WorkflowType(forceReplicationWorkflowName), tag.WorkflowNamespaceID(request.NamespaceID), tag.WorkflowID(we.WorkflowId), tag.WorkflowRunID(we.RunId)}
+
+		resp, err := a.historyClient.DescribeMutableState(ctx, &historyservice.DescribeMutableStateRequest{
```

Comment:
`a.historyClient` -> `a.localHistoryClient`?

Reply:
or source history client

Reply:
i think `historyClient` implicitly mean `local` as other where in the codebase. I can change it if you feel strong about it.